packages/neptune-extension/src/common/utils/path.ts

```diff
@@ -0,0 +1,25 @@
+
+import { PlatformNotebook } from "types/platform";
+
+export async function findNonExistantPath(path: string, platformNotebook: PlatformNotebook) {
+  const parts = path.split('.');
```

Comment:
Remember that `path` is not just a filename, but really path.

If you run jupyter server in some folder `foo` and you will create another `bar` folder inside and a file `my-notebook.ipynb` inside `bar` then Jupyter server will see the path of notebook as `bar/my-notebook.ipynb`.

Assuming you would use `path-browserify` npm module, you could take the file name from path like in this example:
```js
path.basename('/foo/bar/baz/asdf/quux.html');
// Returns: 'quux.html'
```

Reply:
Also remember that there can be dots in folder name.
So better use `node-browserify` rather than calculate filename manually.

Reply:
Sure, we can harden the logic against such cases. However not only jupyter but also neptune server returns the real path, my intention was to recreate it locally. I am not sure which is better, but maybe we should just use a plain filename ? This has this downside, that a notebook will checkout under a different path (root directory) possibly.

Reply:
We can use plain file name.
Just use `node-browserify` to get the filename, so your logic won't fail under such paths like `some.folder/some.file.ipynb`.